feature/text_based.py

```diff
@@ -0,0 +1,724 @@
+             the case that cost metric is None or unicost (for random and greedy)
+
+        Returns
+        -------
+        No return
+       --------
+        * categories is the list of labels. code for labels with categories moved to utils.py
+        """
+        self.matrix = categories.to_numpy()
+        self._num_rows, self._num_cols = self.matrix.shape
+
+        # Content feature (with TextWiser)
+        if method == "random" and cost_metric in [None, "unicost"]:
+            pass
+        elif method == "greedy" and cost_metric == "unicost":
+            pass
+        else:
+            feature_column = self.featurization_method.fit_transform(input_df)
+            self.features = np.array([eval(l) if isinstance(l, str) else l for l in feature_column.tolist()])
+            assert (len(self.features) == self._num_cols), \
+                f"Process Data Error: features size ({len(self.features)}) " \
+                f"should match the number of columns ({self._num_cols})"
+
+        assert (self.matrix.ndim == 2), "Process Data Error: matrix should 2D"
+        if self.selection_size is not None:
+            assert (self.selection_size <= self._num_cols), "Process Data Error: selection_size cannot exceed num columns"
+
+
+
+    def _select_random(self) -> List[int]:# it returns selected column indices
+        # Set the seed
+        random.seed(self.seed)
+        best_selected = []
+        best_covered = 0
+
+        if self.selection_size is None:
+            if self.cost_metric == "unicost":
+                unicost = np.ones(self._num_cols)
+                data = Data(cost=unicost, matrix=self.matrix)
+                unicost_selected = self._solve_set_cover(data)
+                selected_size = len(unicost_selected)
+            else:
+                diversity_cost = self._get_diversity_cost(self._num_cols)
+                data = Data(cost=diversity_cost, matrix=self.matrix)
+                diversity_selected = self._solve_set_cover(data)
+                selected_size = len(diversity_selected)
+        else:
+            selected_size = self.selection_size
+
+        for t in range(self.trials):
+            # Select a sample of selection_size without repetition
+            selected = []
+            while len(selected) < selected_size and self._num_cols is not None:
+                i = random.randint(0, self._num_cols - 1)
+                while i in selected:
+                    i = random.randint(0, self._num_cols -1)
+                selected.append(i)
+
+            # Count covered categories
+            num_row_covered = self._get_num_row_covered(selected)
+            if num_row_covered > best_covered:
+                best_covered = num_row_covered
+                best_selected = selected
+
+        # Calculate coverage metrics
+        num_row_covered = self._get_num_row_covered(best_selected)
+        coverage = num_row_covered / self._num_rows
+
+        if self.verbose:
```

Comment:
I assume the default verbosity will be set to False, so that the user don't see some random printouts..

Reply:
The 'verbose' parameter in the '__init__' method of class ContentSelector is set to 0 by default.